### DISCUSSION TITLE
Where I can found a changelog between releases?

### DISCUSSION THREAD

**Discussion by andrew-demb** [nginxinc/nginx-s3-gateway#201] [Q&A] [answered]
I'm currently using the `20230623` versioned docker image in production and want to update it to the latest (`20240113`).

I can't find GitHub releases for the repository (https://github.com/nginxinc/nginx-s3-gateway/releases), annotated tags (https://github.com/nginxinc/nginx-s3-gateway/tags), description for package container image (https://github.com/nginxinc/nginx-s3-gateway/pkgs/container/nginx-s3-gateway%2Fnginx-oss-s3-gateway/166705090?tag=latest-20240113) or changelog file in the repository.

What can I use to know what's changed between the two releases?
Can I consider all "latest"-named images as stable?

Full image: `ghcr.io/nginxinc/nginx-s3-gateway/nginx-oss-s3-gateway:latest-20230623`

**Comment by 4141done** [comment, marked as answer]
Hello,

Thanks for raising this. This project started out a little experimental but now as we're seeing more interest it seems to be time to formalize some things like changelogs and versioning.

In terms of answering your question - we don't have a changelog at the moment but we are consistent about squashing our commits on merge so the commit log on `master` should give you an idea of changes.

```
git log --since="2023-06-23" --pretty=format:"* %s\n"
* fix and issue where final build cannot find the base image for build (#200)\n
* Build multi-architecture images (#197)\n
* fix: handle IMDSv2 auth when checking if we are running on EC2 instance (#193)\n
* feat: Add option to specify a prefix to be prepended to object paths (#190)\n
* feat: Add new configuration option STRIP_LEADING_DIRECTORY_PATH. (#185)\n
* FIX: update NGINX, NGINX Plus, NJS, and NGINX XSLT module to newest versions (#184)\n
* fix: Add x-amz-content-sha256 to CanonicalHeaders (#179)\n
* feature: jsdoc annotations and generation (#177)\n
* fix: gracefully handle comments in resolv.conf. Fixes #171 (#176)\n
* Fixes #164 (#167)\n
* fix: error messages for missing dependencies, allow test suite to run on MacOS by supporting `md5` (#162)\n
* chore: small fixes to test scripts and README for clarity (#161)\n
* Add missing CORS headers for 404 status code\n
* Fixed directory listing prefixing\n
* Change Minio CLI download to use --insecure\n
* Make allow-directory-listing and provide-index-page compatible (#139)\n
* Fix mc download in GitHub Action\n
* Added installation script running example\n
* Fixed listing xslt template and removed auto_envsubst_on_include\n
```

If I strip out things I know to be administrative it's mostly addition of new config options.  We have not yet shipped an intentionally breaking change nor would we do so without a proper versioning change to make sure we don't break people.

Hope that helps.  I'll create issues for these items so that the public can know the plan.
